src/lkl/syscall-overrides-fstat.c

```diff
@@ -0,0 +1,56 @@
+#include <sys/stat.h>
+#include <lkl.h>
+#include <lkl_host.h>
+#include <lkl/asm/stat.h>
+#include <string.h>
+
+#include "lkl/syscall-overrides-fstat.h"
+#include "enclave/enclave_util.h"
+
+static void copy_lkl_stat_to_user(struct lkl_stat *lkl_stat, struct stat *stat) {
+	stat->st_dev = lkl_stat->st_dev;
+	stat->st_ino = lkl_stat->st_ino;
+	stat->st_mode = lkl_stat->st_mode;
+	stat->st_nlink = lkl_stat->st_nlink;
+	stat->st_uid = lkl_stat->st_uid;
+	stat->st_gid = lkl_stat->st_gid;
+	stat->st_rdev = lkl_stat->st_rdev;
+	stat->st_size = lkl_stat->st_size;
+	stat->st_blksize = lkl_stat->st_blksize;
+	stat->st_blocks = lkl_stat->st_blocks;
+	stat->st_atim.tv_sec = lkl_stat->lkl_st_atime;
+	stat->st_atim.tv_nsec = lkl_stat->st_atime_nsec;
+	stat->st_mtim.tv_sec = lkl_stat->lkl_st_mtime;
+	stat->st_mtim.tv_nsec = lkl_stat->st_mtime_nsec;
+	stat->st_ctim.tv_sec = lkl_stat->lkl_st_ctime;
+	stat->st_ctim.tv_nsec = lkl_stat->st_ctime_nsec;
+}
+
+long syscall_fstat_override(int fd, struct stat* stat) {
+    struct lkl_stat lkl_stat = {0};
+    long ret;
+
+    if (orig_fstat == NULL)
+        sgxlkl_fail("Error: fstat syscall handler not defined");
+
+    ret = orig_fstat(fd, &lkl_stat);
+    if (ret == 0)
+      copy_lkl_stat_to_user(&lkl_stat, stat);
+
+    return ret;
+}
+
+long syscall_newfstatat_override(int dfd, const char *fn, struct stat *stat, int flag) {
+    struct lkl_stat lkl_stat = {0};
+    long ret;
+
+    if (orig_newfstatat == NULL)
+        sgxlkl_fail("Error: newfstatat syscall handler not defined");
+
+    ret = orig_newfstatat(dfd, fn, &lkl_stat, flag);
```

Comment:
This function pointer needs to be cast to the correct type (it is UB to call a C function with a different prototype).